rfcs/build-system/04-build-shield.md

```diff
@@ -0,0 +1,107 @@
+# RFC: Establish "build shield" rotation
+
+## Summary
+
+We need a designated build "firefighter" in each time zone who's responsible for addressing any urgent build-related issues and keeping things unblocked.
+
+This is a separate rotation from regular shield, and at least with the initial set of responsibilities proposed, it's a much lower average time commitment.
+
+## Background/Problem
+
+To this point, build "firefighting" tends to de facto fall to one or two particular people in each time zone, which has multiple disadvantages:
+
+- Consistent, disproportionate negative impact to these people's "official" work
+- Lack of recognition/appreciation for time spent and value provided to the team
+- Build-related knowledge stays concentrated with a very limited number of people
+- Harder to recognize patterns of failures
+
+Having build shield as a desginated, rotating responsibility will help address all of these issues. It will also help more people on the team gain experience with build troubleshooting.
+
+## Detailed Design or Proposal
+
+### Who
+
+Initially this will rotate within the build v-team, but once we have better documentation (see later in doc) we may be able to expand it to the team as a whole.
+
+Since the focus is addressing urgent issues, we need **one person per time zone.**
+
+### Time commitment
+
+Shift duration will probably be either 1 or 2 weeks (need to discuss).
+
+Total time commitment would be allocated as **up to 25%**. _(DISCUSS: is this enough?)_
+
+Some weeks you'll get lucky and have almost nothing to do, but other weeks something will "blow up" in a way that requires much more time to address. If the time goes beyond ~25% _(or whatever number we decide above)_ it's worth calling out to management and/or pulling in other people as needed.
```

Comment:
Do we want folks to spend any extra time improving docs or fixing longer term issues? Or should we instead schedule that work specifically, what do folks think?

Reply:
What I had in mind was that if you encounter an issue and how to address it isn't already documented, it's good if you can document that. More general documentation or long-term improvements would be out of scope (unless the person had extra time and really wanted to work on one of those things; or we could consider scheduling such work to overlap with a person's build shield shift depending on the shift length).